runtime/lua/vim/lsp/util.lua

```diff
@@ -44,10 +44,20 @@ local function get_border_size(opts)
     height = 2
     width = 2
   else
-    height = height + vim.fn.strdisplaywidth(border[2][1])  -- top
-    height = height + vim.fn.strdisplaywidth(border[6][1])  -- bottom
-    width  = width  + vim.fn.strdisplaywidth(border[4][1])  -- right
-    width  = width  + vim.fn.strdisplaywidth(border[8][1])  -- left
+    local function border_width(id)
+      if type(border[id]) == "table" then
+        -- border specified as a table of <character, highlight group>
+        return vim.fn.strdisplaywidth(border[id][1])
+      elseif type(border[id]) == "string" then
+        -- border specified as a list of border characters
+        return vim.fn.strdisplaywidth(border[id])
+      end
+      error("floating preview border is not correct. Please refer to the docs |vim.api.nvim_open_win()|" .. vim.inspect(border))
+    end
+    height = height + border_width(2)  -- top
```

Comment:
`strdisplaywidth` shouldn't be used for height, just whether there is a char (1) or not (0).

Reply:
Good point. I actually wonder, can you have a top border with something like `#\n#` that would be two then?

Reply:
fixed